db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java

```diff
@@ -0,0 +1,73 @@
+package com.eventstore.dbclient;
+
+import com.eventstore.dbclient.proto.persistentsubscriptions.Persistent;
+import com.eventstore.dbclient.proto.persistentsubscriptions.PersistentSubscriptionsGrpc;
+import io.grpc.Metadata;
+import io.grpc.stub.MetadataUtils;
+
+import java.util.concurrent.CompletableFuture;
+
+public abstract class AbstractUpdatePersistentSubscription {
+    private final GrpcClient connection;
+    private final String group;
+    private final AbstractPersistentSubscriptionSettings settings;
+    private Metadata metadata;
+
+    public AbstractUpdatePersistentSubscription(GrpcClient connection, String group,
+                                                AbstractPersistentSubscriptionSettings settings, Metadata metadata) {
+        this.connection = connection;
+        this.group = group;
+        this.settings = settings;
+        this.metadata = metadata;
+    }
+
+    protected Persistent.UpdateReq.Settings.Builder createSettings() {
+        return Persistent.UpdateReq.Settings.newBuilder();
+    }
+
+    protected abstract Persistent.UpdateReq.Options.Builder createOptions();
+
+    public CompletableFuture execute() {
+        return this.connection.run(channel -> {
+            CompletableFuture result = new CompletableFuture();
+            PersistentSubscriptionsGrpc.PersistentSubscriptionsStub client = MetadataUtils
+                    .attachHeaders(PersistentSubscriptionsGrpc.newStub(channel), metadata);
+            Persistent.UpdateReq.Settings.Builder settingsBuilder = createSettings();
+
+            settingsBuilder
+                    .setResolveLinks(settings.isResolveLinks())
+                    .setReadBatchSize(settings.getReadBatchSize())
+                    .setMinCheckpointCount(settings.getMinCheckpointCount())
+                    .setMaxCheckpointCount(settings.getMaxCheckpointCount())
+                    .setMessageTimeoutMs(settings.getMessageTimeoutMs())
+                    .setMaxSubscriberCount(settings.getMaxSubscriberCount())
+                    .setMaxRetryCount(settings.getMaxRetryCount())
+                    .setLiveBufferSize(settings.getLiveBufferSize())
+                    .setHistoryBufferSize(settings.getHistoryBufferSize())
+                    .setExtraStatistics(settings.isExtraStatistics())
+                    .setCheckpointAfterMs(settings.getCheckpointAfterMs());
+
+            switch (settings.getStrategy()) {
```

Comment:
**CONCERN:** Aren't we missing `PinnedByCorrelation` here?

Reply:
Think this is not implemented yet, can't find this on the server ?

Reply:
PinnedByCorrelation isn't available on the server proto yet, so should come in a later PR

Reply:
It might be worth creating an issue for that to not forget about it.

Reply:
Done https://github.com/EventStore/EventStoreDB-Client-Java/issues/88